build.gradle.kts

```diff
@@ -9,55 +8,51 @@ plugins {
     id("org.jlleitschuh.gradle.ktlint") apply true
     id("com.google.dagger.hilt.android") version "2.48.1" apply false
 }
+val localProperties = Properties()
+if (rootProject.file("local.properties").exists()) {
+    localProperties.load(FileInputStream(rootProject.file("local.properties")))
+}
 
-rootProject.ext {
-    this.set("appId", "uk.gov.onelogin")
-    this.set("compileSdkVersion", 34)
-    this.set("configDir", "${rootProject.rootDir}/config")
-    this.set("minSdkVersion", 29)
-    this.set("targetSdkVersion", 34)
-
-    val localProperties = Properties()
-    if (rootProject.file("local.properties").exists()) {
-        localProperties.load(FileInputStream(rootProject.file("local.properties")))
+setProperty("appId", "uk.gov.onelogin")
+setProperty("compileSdkVersion", 34)
+setProperty("configDir", "${rootProject.rootDir}/config")
+setProperty("minSdkVersion", 29)
+setProperty("targetSdkVersion", 34)
+setProperty("versionCode", getVersionCode())
+setProperty("versionName", getVersionName())
+
+fun getVersionCode(): Int {
+    val code: Int = if (rootProject.hasProperty("versionCode")) {
+        rootProject.property("versionCode").toString().toInt()
+    } else if (localProperties.getProperty("versionCode") != null) {
+        localProperties.getProperty("versionCode").toString().toInt()
+    } else {
+        throw Error(
+            "Version code was not found as a command line parameter or a local property"
+        )
```

Comment:
Do we have any documentation somewhere around the potential next steps when encountering this error?

Reply:
I'm not sure we do, something to look into I suppose